src/main/java/com/synopsys/integration/alert/channel/azure/boards/service/AzureTransitionHandler.java

```diff
@@ -0,0 +1,84 @@
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.channel.azure.boards.service;
+
+import java.util.List;
+import java.util.Optional;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.synopsys.integration.alert.common.channel.issuetracker.service.TransitionHandler;
+import com.synopsys.integration.azure.boards.common.model.AzureArrayResponseModel;
+import com.synopsys.integration.azure.boards.common.service.state.AzureWorkItemTypeStateService;
+import com.synopsys.integration.azure.boards.common.service.state.WorkItemTypeStateResponseModel;
+import com.synopsys.integration.azure.boards.common.service.workitem.AzureWorkItemService;
+import com.synopsys.integration.azure.boards.common.service.workitem.response.WorkItemResponseFields;
+import com.synopsys.integration.azure.boards.common.service.workitem.response.WorkItemResponseModel;
+import com.synopsys.integration.azure.boards.common.util.AzureFieldsExtractor;
+import com.synopsys.integration.exception.IntegrationException;
+
+public class AzureTransitionHandler implements TransitionHandler<WorkItemTypeStateResponseModel> {
+    public static final String WORK_ITEM_STATE_CATEGORY_PROPOSED = "Proposed";
+    public static final String WORK_ITEM_STATE_CATEGORY_IN_PROGRESS = "InProgress";
+    public static final String WORK_ITEM_STATE_CATEGORY_RESOLVED = "Resolved";
+    public static final String WORK_ITEM_STATE_CATEGORY_COMPLETED = "Completed";
+
+    private final AzureBoardsProperties azureBoardsProperties;
+    private final AzureWorkItemService azureWorkItemService;
+    private final AzureWorkItemTypeStateService azureWorkItemTypeStateService;
+    private final AzureFieldsExtractor azureFieldsExtractor;
+
+    public AzureTransitionHandler(Gson gson, AzureBoardsProperties azureBoardsProperties, AzureWorkItemService azureWorkItemService, AzureWorkItemTypeStateService azureWorkItemTypeStateService) {
+        this.azureBoardsProperties = azureBoardsProperties;
+        this.azureWorkItemService = azureWorkItemService;
+        this.azureWorkItemTypeStateService = azureWorkItemTypeStateService;
+        this.azureFieldsExtractor = new AzureFieldsExtractor(gson);
+    }
+
+    @Override
+    public String extractTransitionName(WorkItemTypeStateResponseModel workItemTypeState) {
+        return workItemTypeState.getName();
+    }
+
+    @Override
+    public List<WorkItemTypeStateResponseModel> retrieveIssueTransitions(String workItemIdString) throws IntegrationException {
+        String organizationName = azureBoardsProperties.getOrganizationName();
+        Integer workItemId = Integer.parseInt(workItemIdString);
+        WorkItemResponseModel workItem = azureWorkItemService.getWorkItem(organizationName, workItemId);
+
+        JsonObject workItemFields = workItem.getFields();
```

Comment:
When we have chains of calls like this, we should try to have a method in the original object that will directly give us the final result for convenience. I don't see workItem used anywhere else.

Reply:
Yeah, I was definitely thinking about it. You pointing this out has convinced me that it needs to be done.